Apps/Database/src/Delegates/DBWalletDelegate.cs

```diff
@@ -0,0 +1,212 @@
+﻿// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Database.Delegates
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Text.Json;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    [ExcludeFromCodeCoverage]
+    public class DBWalletDelegate : IWalletDelegate
+    {
+        private readonly ILogger logger;
+        private readonly GatewayDbContext dbContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBWalletDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="dbContext">The context to be used when accessing the database.</param>
+        public DBWalletDelegate(
```

Comment:
Where would we check to make sure the associated resource for a credential belongs to the associated patient?

Reply:
The service layer.